[CS2113T-T09-1] Sherpass

jltha opened the pull request.

SeanHoWB commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+Below is a sequence diagram of what happens 
+as the user wishes to see the schedule (timetable) for 25th May 2022:
+
+![](images/showScheduleForADate.png)
```
This image hard to see, will be better if you can use ref

SeanHoWB commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+
+Sequence diagram for `Timer` when user starts and stops a timer:
+
+![TimerClassSD](https://user-images.githubusercontent.com/69501969/160768104-fa7e06e3-1be8-4387-b75d-ae4e79bca5b7.png)
```
is update command here a method? if so need to do the proper self invoke syntax, arrow itself, activate, unarrow itself, deactivate. Same as start

SeanHoWB replied:
![image](https://user-images.githubusercontent.com/69450377/160962525-34390ca7-2152-41ba-b362-c2839b3ce441.png)
what causes this? what happens here after the command is ran? Its abit unclear from the diagram

SeanHoWB commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+
+Timetable component consists of `Timer`, `Stopwatch`, `Countdown`, `TimerParser`, `TimerLogic` and various commands.
 
 ## Design & implementation
```
i suggest splitting design and implementation, where design can include stuff like architecture overview,class diagrams of the 4 major compnents above

Yzkkk commented on docs/puml/showScheduleForADate.puml:
```diff
@@ -0,0 +1,73 @@
+
+hide footbox
+
+participant User as User
```
You have changed a colour for sequence diagram under timetable implementation, which is good and its clear to the reader that it falls under a different category. However, for the sequence diagram under the storage implementation section, you are still using red, so this may be confuse the reader and defeats the purpose of having colours in diagrams.

SeanHoWB reviewed (commented):
Overall looks good, but more can be added to the project in terms of diagrams and elaboration of different components

Yzkkk commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+`callStopTimer` method is a call to a method in `Countdown` to stop the timer. Control goes back to the user for 
+further commands.
+
+#### Design considerations for Timer class
+- Current implementation: Create `Timer` from scratch, using the sleep function of threads to keep 
+track of time
+  - Pros: Same overhead of needing to track the time left of the timer
+  - Pros: No need to follow Java’s `Timer` class syntax, which can be confusing at times
+  - Pros: Implementation is simple and straight-forward
+  - Cons: Have to manage how we interrupt the thread after stopping the timer
+- Alternative: Using Java's `Timer` class
+  - Pros: The way of keeping track of the time has already been implemented
+  - Pros: Using a standard library usually makes the program less prone to various errors
+  - Cons: Still have to implement a way to keep track of time for our purposes of pausing a timer, since the library 
+  provided by Java has no way of pausing the timer, only stopping it.
+
```
Nice!

Yzkkk commented on docs/puml/showScheduleForADate.puml:
```diff
@@ -0,0 +1,73 @@
+hide footbox
+
+participant User as User
+participant Main as Main
```
The diagram is hard to read, I suggest using ref and extract the interaction between Timetable and TaskList.

Yzkkk commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+- [Introduction](#introduction)
+- [Architecture](#architecture)
+- [Design & Implementation](#design--implementation)
+  - [Study Session Implementation](#study-session-implementation)
+  - [Timetable Implementation](#timetable-implementation)
+  - [Storage Implementation](#storage-implementation)
```
I suggest including the sub-headings as well, eg Timer Implementation, Study session usage scenario, Design considerations for Timer class to be under Study Session Implementation.

angyongming commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+Below is a sequence diagram of what happens 
+as the user wishes to see the schedule (timetable) for 25th May 2022:
+
+![](images/showScheduleForADate.png)
```
I agree with sean, diagram is rather complicated and I think it would be better to use a reference frame and split this diagram into multiple diagrams to see each one clearer.

angyongming replied:
Following up from my comment of the previous diagram, the Main class seems to be reading the userInput?

angyongming commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+
+The loading of a save file is done with the function
+
+`Storage#load()` - Loads a saved JSON file and returns an ArrayList of task
+
+The path of the JSON file is provided as a parameter in the constructor of `Storage` hence 
+there is no need for any parameters in the `Storage#load()`. Since a save file will be created in the 
```
Not sure if there is a formatting error here or if it was intentional but why is there a # betwen Storage and its load() method? Is it supposed to be Storage.load() in this case?

angyongming commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+- Text file with custom format
+  - Pros: Might use less space (negligible)
+  - Cons: Unintuitive for users to modify
 ## Product scope
 ### Target user profile
```
Take note that there is a minor formatting error here:
![image](https://user-images.githubusercontent.com/69451153/160969414-d0697387-127c-48f8-a5df-40542bc90941.png)

angyongming commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+
+Sequence diagram for `Timer` when user starts and stops a timer:
+
+![TimerClassSD](https://user-images.githubusercontent.com/69501969/160768104-fa7e06e3-1be8-4387-b75d-ae4e79bca5b7.png)
```
Since your project has a UI class, shouldn't the UI class be responsible for taking in the userInput instead of the StudyCommand class?

angyongming replied:
Just saw the rest of the diagrams, I noticed a little bit of inconsistency here whereby StudyCommand takes in userInput for this diagram, but for the other diagram it is Main, and for another diagram it is the UI class

angyongming commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+
+The sequence diagram of `Storage#handleCorruptedSave()` is shown here:
 
+![](images/StorageCorruptedSD.png)
```
Following up on my comment on the previous diagram, the reading of userInput here is done by Ui class, perhaps would be better to standardize this across the other diagrams!

angyongming commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,314 @@
+- Timetable
+- Study session
+
+### UI
+
+UI component consists the `Ui` class which manages interaction (receiving inputs and showing outputs) between the user 
+and the application.
+
+### Storage
+
+Storage component consists `Storage` and `StorageParser` classes.
+`Storage` class handles loading, writing and saving
+data to and from a JSON file, such that users' data will be saved automatically.
+`StorageParser` class handles the
+parsing of JSON from the saved data file.
+
+### Timetable
+
+For components with more complicated use-cases (`Task` and `Timer`), we separate an extra Logic class to achieve better
+modularity, such that each class addresses a separate concern.
+
+Timetable component consists of `Timetable`, `Task`, `TaskList`, `TaskParser`, `TaskLogic` and various commands.
+
+### Study session
+
+Timetable component consists of `Timer`, `Stopwatch`, `Countdown`, `TimerParser`, `TimerLogic` and various commands.
```
Very clear overview of the 4 main components of the programme. Good job!

angyongming reviewed (commented):
Good use of visuals and sufficient use of examples! Explanations are clear and concise and the overall flow is very coherent and readable. Nice job :)

okkhoy commented on src/main/java/seedu/sherpass/command/AddCommand.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    /**
+     * Accept parsed user input for by date, in proper format.
```
Accepts ...

same thing across the comments

okkhoy commented on src/main/java/seedu/sherpass/command/Command.java:
```diff
@@ -0,0 +1,11 @@
+
+public abstract class Command {
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage);
```
This method needs a header comment as it is an abstract method, meaning you are forcing someone else to provide an implementation, and they need to know what to implement

okkhoy commented on src/main/java/seedu/sherpass/command/ShowCommand.java:
```diff
@@ -0,0 +1,176 @@
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        if (dayInput != null) {
+            ui.showToUser("Here is the schedule you wanted:");
+            Timetable.showScheduleByDay(dayInput, taskList, ui);
+            return;
+        }
```
good use of guard clause 👍

okkhoy commented on src/main/java/seedu/sherpass/command/ShowCommand.java:
```diff
@@ -0,0 +1,176 @@
+            Timetable.showNovemberSchedule(taskList, ui);
+            ui.showToUser(WAITING_FOR_USER_NEXT_INPUT);
+            break;
+        case "decemeber":
+        case "dec":
```
would lead to problems where the user doesn't use 3-letter/full name, e.g., decem

yuzhongng replied:
Would it be fine if we make the 3-letter form very clear in the user guide?

okkhoy commented on src/main/java/seedu/sherpass/command/StudyCommand.java:
```diff
@@ -0,0 +1,59 @@
+
+    /**
+     * Method is called when user chooses to enter Study mode. User is able to start, pause and stop a timer in Study
+     * mode. Only one timer can be running at a time. User can leave Study mode by typing "leave".
```
doesn't follow the standard for header comments

okkhoy commented on src/main/java/seedu/sherpass/constant/CommandParameter.java:
```diff
@@ -0,0 +1,11 @@
+package seedu.sherpass.constant;
+
+public class CommandParameter {
+    public static final String DO_DATE_DELIMITER = "/do";
+    public static final String BY_DATE_DELIMITER = "/bydate";
+    public static final String BY_TIME_DELIMITER = "/bytime";
+    public static final String FREQUENCY_DELIMITER = "/repeat";
+    public static final String START_TIME_DELIMITER = "/start";
+    public static final String END_TIME_DELIMITER = "/end";
+    public static final String TIMER_CUSTOM_DELIMITER = "/custom";
+}
```
1. There are other constants also declared in other java files. Think what makes these special that they should be in a separate file (or the other way round).
2. The names don't follow the suggested convention

okkhoy commented on src/main/java/seedu/sherpass/constant/Index.java:
```diff
@@ -0,0 +1,28 @@
+package seedu.sherpass.constant;
+
+public class Index {
+    public static final int OPTIONS_INDEX = 0;
+    public static final int HELP_OPTIONS_INDEX = 1;
+    public static final int DIRECTORY_INDEX = 0;
+    public static final int STUDY_COMMAND_INDEX = 0;
+    public static final int STUDY_PARAMETER_INDEX = 1;
+    public static final int TIMER_FORMAT_INDEX = 1;
+    public static final int DEFAULT_TIMER_INDEX = 1;
+    public static final int CUSTOM_TIMER_INDEX = 1;
+    public static final int DEFAULT_TIMER_ONE = 1800;
+    public static final int DEFAULT_TIMER_TWO = 3600;
+    public static final int DEFAULT_TIMER_THREE = 5400;
+    public static final int DEFAULT_TIMER_ZERO = 30;
+    public static final int INDENT_FACTOR = 4;
+    public static final int CUSTOM_COMMAND_INDEX = 0;
+    public static final int SHOW_OPTION_INDEX = 1;
+    public static final int ZERO_INDEX_OFFSET = 1;
+    public static final int WHITESPACE_OFFSET = 1;
+    public static final int START_OF_STRING = 0;
+    public static final int SPLIT_TWO_PART_LIMIT = 2;
+    public static final int SPLIT_FIRST_PART_INDEX = 0;
+    public static final int COMMAND_CONTENT_INDEX = 1;
+    public static final int EDIT_INDEX = 0;
+    public static final int EDIT_TASK_CONTENT = 1;
+    public static final int INDEX_OFFSET = 1;
+}
```
Constant names don't follow the convention

jltha replied:
Hi Prof, may I know for which part is the convention being violated? The [link](https://se-education.org/guides/conventions/java/basic.html#naming) to the coding standard only mentions that underscores must be used to separate words, and all words must be in uppercase.

okkhoy commented on src/main/java/seedu/sherpass/constant/Message.java:
```diff
@@ -0,0 +1,99 @@
+package seedu.sherpass.constant;
+
+public class Message {
```
Why not add all other such constants here and consolidate?

okkhoy commented on src/main/java/seedu/sherpass/exception/InvalidInputException.java:
```diff
@@ -0,0 +1,12 @@
+
+public class InvalidInputException extends Exception {
+
+    public InvalidInputException(String message) {
```
need header comments as you are defining your own exception. at least need to tell what it is

okkhoy commented on src/main/java/seedu/sherpass/task/Task.java:
```diff
@@ -0,0 +1,202 @@
+    protected int index;
+
+    /**
+     * Creates a constructor for the parent class of tasks, 'Task'.
```
creates an object ...

okkhoy commented on src/main/java/seedu/sherpass/task/TaskList.java:
```diff
@@ -0,0 +1,508 @@
+            return currentDate.plusWeeks(1);
+        }
+        return currentDate.plusMonths(1);
+    }
```
why is increment date inside task list?

okkhoy commented on src/main/java/seedu/sherpass/task/TaskList.java:
```diff
@@ -0,0 +1,508 @@
+import static seedu.sherpass.constant.Message.NEWLINE;
+import static seedu.sherpass.constant.Message.TAB_INDENT;
+
+public class TaskList {
```
I think this class violates SRP. please see if you can refactor

okkhoy commented on src/main/java/seedu/sherpass/task/TaskList.java:
```diff
@@ -0,0 +1,508 @@
+     * @throws TimeClashException    if the new do date has is clashing with other tasks
+     * @throws InvalidInputException if the dates are invalid e.g. start date after end date
+     */
+    public void editSingleTaskContent(int editIndex, String taskDescription,
```
This could easily belong to the Task class rather than task list as you are working with one task

okkhoy commented on src/main/java/seedu/sherpass/task/TaskList.java:
```diff
@@ -0,0 +1,508 @@
+    /**
+     * Returns
+     * Printed tasks applies to non-recurring tasks.
+     *
```
violates convention

okkhoy commented on src/main/java/seedu/sherpass/timer/Timer.java:
```diff
@@ -0,0 +1,65 @@
+import seedu.sherpass.util.Ui;
+
+
+public abstract class Timer extends Thread {
```
All abstract methods in this class need header comments

okkhoy commented on src/main/java/seedu/sherpass/timer/TimerLogic.java:
```diff
@@ -0,0 +1,317 @@
+
+
+    /**
+     * Call method when window is minimised.
```
calls ... 
same for the other methods

okkhoy commented on src/main/java/seedu/sherpass/timetable/Timetable.java:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    /**
+     * Generate a timetable for tomorrow.
```
Generates ...

okkhoy commented on src/test/java/seedu/sherpass/util/parser/ParserTest.java:
```diff
@@ -0,0 +1,4 @@
+package seedu.sherpass.util.parser;
+
+public class ParserTest {
+}
```
??

okkhoy reviewed (commented):
In general, the code is quite neat! 
I have highlighted some potential issues that may need fixing.
My suggestion is for each of you to take ownership and fix the files that you guys wrote. That way, each of you can get credit for good code 😄

thedarie committed:
Add period in comment to fix checkstyle error

thedarie committed:
Update UML diagrams

thedarie committed:
Add some test cases for manual testing

thedarie committed:
Fix bug when updating task

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

laiisaac committed:
Merge branch 'master' into isaac-BugFixes

yuzhongng committed:
Merge pull request #150 from jltha/branch-v2.1-Fixes

Jun Lim ped fixes

thedarie committed:
Update task section in developer guide

BradenTeo committed:
add JUnit test for Clear Command
2 tests are omitted from final build as they incur an error when
run with other tests, but are able to run individually

BradenTeo committed:
fix Checkstyle issue

thedarie committed:
Merge branch 'master' into daryl-ImproveStorage

# Conflicts:
#	src/main/java/seedu/sherpass/task/TaskList.java

BradenTeo committed:
Merge branch 'master' into branch-ClearExpiredOrDone

thedarie committed:
Update screenshots to match new output format

BradenTeo committed:
fix JUnit issue when loading the data/test.json file

BradenTeo committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

BradenTeo committed:
Merge branch 'branch-ClearExpiredOrDone' of
https://github.com/BradenTeo/tp into branch-ClearExpiredOrDone

BradenTeo committed:
Fix more Checkstyle issues

BradenTeo committed:
Add newline to the end of some files

thedarie committed:
Merge branch 'master' into daryl-DG

# Conflicts:
#	docs/UserGuide.md

thedarie committed:
Update sequence diagram for storage

thedarie committed:
Update PPP

thedarie committed:
Update checking of deadline

yuzhongng committed:
Add javadocs for the constructor and method

yuzhongng committed:
Add javadoc for various Timetable methods

yuzhongng committed:
Add javadoc for one of the TimetableLogic method

yuzhongng committed:
Add javadoc for TimetableParser methods

yuzhongng committed:
Add TimetableParserTest Junit Test to test the method given by TimetableParser Class

laiisaac committed:
Update Timer diagrams in Developer guide

laiisaac committed:
Update javadocs for abstract methods

laiisaac committed:
Update glossary for ambigious terms in study session for DG

BradenTeo committed:
Move string constants from command classes to a new class in the
constant package

BradenTeo committed:
Merge pull request #151 from thedarie/daryl-ImproveStorage

Daryl Refactor

yuzhongng committed:
Update User Guide for Add Command with optional deadline input and Timetable Section

yuzhongng committed:
Replace the original photo to a black blackground

yuzhongng committed:
Update the help add command to reflect the latest changes in the features

yuzhongng committed:
Update help command to comply with checkstyle

laiisaac committed:
Add JUnit tests for Timer

laiisaac committed:
Update TimerTest

laiisaac committed:
Disable TimerTest to allow build to succeed

BradenTeo committed:
Merge pull request #152 from BradenTeo/branch-ClearExpiredOrDone

Braden Update clear command to delete expired or completed tasks

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

laiisaac committed:
Merge branch 'master' into isaac-BugFixes

yuzhongng committed:
Merge branch 'master' into yuzhong-Timetable

# Conflicts:
#	src/main/java/seedu/sherpass/command/AddCommand.java

thedarie committed:
Merge branch 'master' into daryl-DG

# Conflicts:
#	docs/UserGuide.md

yuzhongng committed:
Update Message_Usage_add to reflect the latest features

yuzhongng committed:
Update javadoc for showCommand constructor

thedarie committed:
Update image to remove circle icon

BradenTeo committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

thedarie committed:
Change format of method names to be clearer

BradenTeo committed:
Add PPP

laiisaac committed:
Update TimerTest

laiisaac committed:
Add parameters to build.gradle to fix TimerTest not working

laiisaac committed:
Update build.gradle to fix TimerTest issues

laiisaac committed:
Disable TimerTest as issues were not able to be resolved

jltha committed:
Merge pull request #154 from laiisaac/isaac-BugFixes

Isaac V2.1 improvements

thedarie committed:
Merge branch 'master' into daryl-DG

laiisaac committed:
Merge pull request #155 from BradenTeo/branch-PPP

Braden PPP

laiisaac committed:
Merge pull request #145 from thedarie/daryl-DG

Daryl documentations

yuzhongng committed:
Update User Guide for editing of task

yuzhongng committed:
Add a warning icon at the warning line

yuzhongng committed:
Update User Guide for Timetable

Make it explictly clear that showing a specfic month timetable will only accept 2 kind of inputs

yuzhongng committed:
Replace photos with black blackground

yuzhongng committed:
Update help command for editing of task

laiisaac committed:
Update timer screenshots in user guide

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

yuzhongng committed:
Merge branch 'master' into yuzhong-Timetable

# Conflicts:
#	docs/UserGuide.md

yuzhongng committed:
Update Command Summary in the UserGuide for timetable features

yuzhongng committed:
Update the correct file path for the picture

yuzhongng committed:
Replace the original photo with a black background

yuzhongng committed:
Replace the original photo with a black background

yuzhongng committed:
Replace the original photo with a black background

yuzhongng committed:
Update UserGuide for the show any specific month command

laiisaac committed:
Add feature where timer prints time remaining/elapsed at regualr intervals

Print statements at regular intervals were removed when Timer was refactored into an abstract class, feature has now been re-added

yuzhongng committed:
Update User Guide for the timetable section

yuzhongng committed:
Update YuZhong PPP

yuzhongng committed:
Update DeveloperGuide for Timetable

thedarie committed:
Merge pull request #153 from yuzhongng/yuzhong-Timetable

Yuzhong timetable

yuzhongng committed:
Update yuzhong PPP

yuzhongng committed:
Update AboutUs for the portfolio

yuzhongng committed:
Merge branch 'yuzhong-Timetable' into yuzhong-Documentation

jltha committed:
Resolve merge conflicts

jltha committed:
Merge branch 'master' into branch-v2.1-docs

# Conflicts:
#	docs/UserGuide.md

yuzhongng committed:
Update spelling error

jltha committed:
Update doc for checkDateTimeClash method in TaskUtil class

jltha committed:
Remove unused imports in TaskList class

jltha committed:
Fix bug in editing task(s)

Occur when editing just the by date and by time (bug)

jltha committed:
Edit junit test in ClearCommandTest

Disable tests instead of having gradle filter the tests

jltha committed:
Update junit tests in TaskListTest after updating the edit feature

jltha committed:
Add junit test for testing editing only the bydate and bytime

jltha committed:
Add back filtering in build.gradle for ClearCommandTest class

jltha committed:
Disable certain junit tests

jltha committed:
Create extra json file to test edit feature

jltha committed:
Update testEdit.json after running junit test

jltha committed:
Add javadoc for execute method in Command class

jltha committed:
Update timetable section in UG

jltha committed:
Add parameter table in clear command (in UG)

yuzhongng committed:
Add equals method for AddCommand object

yuzhongng committed:
Add J unit test for Parser Class

yuzhongng committed:
Merge pull request #158 from jltha/branch-v2.1-Fixes

Jun Lim Fix bug in editing only the deadline date and time

yuzhongng committed:
Merge branch 'master' into yuzhong-Documentation

jltha committed:
Add more pictures for adding, editing, and deleting of tasks in UG

jltha committed:
Remove unused image

jltha committed:
Remove unused imports in TaskList

jltha committed:
Edit javadoc for enterStudyMode in StudyCommand

jltha committed:
Fix typo in javadoc for ShowCommand

BradenTeo committed:
Reformat help messages

BradenTeo committed:
Refactor string constants in Ui class

BradenTeo committed:
Minor edits to PPP

BradenTeo committed:
Add 'Help' and 'Bye' sections to UG
Add other minor fixes, including shifting the 'Saving your tasks'
section to further down in the UG to correspond to the table of
contents

yuzhongng committed:
Add colon to represent the class object

yuzhongng committed:
Add colon to represent the class object

yuzhongng committed:
Update Developer Guide for Timetable testing

yuzhongng committed:
Update table of content in Developer Guide

laiisaac committed:
Merge pull request #156 from yuzhongng/yuzhong-Documentation

Yuzhong documentation and J-Unit test

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

laiisaac committed:
Merge branch 'master' into isaac-BugFixes

laiisaac committed:
Update Isaac PPP

jltha committed:
Update timetable section for DG

jltha committed:
Remove unneeded line in DG (under Timetable section)

thedarie committed:
Merge pull request #160 from laiisaac/isaac-BugFixes

Isaac v2.1 improvements

jltha committed:
Update DG

1) Update timetable component in intro
2) Update architecture diagram

jltha committed:
Update PPP

jltha committed:
Merge branch 'master' into branch-v2.1-docs

# Conflicts:
#	docs/puml/ShowScheduleForADateSDOne.puml
#	docs/puml/showScheduleForADateTwo.puml
#	src/main/java/seedu/sherpass/task/TaskList.java

jltha committed:
Update PPP

jltha committed:
Add @@author tags in TaskUtil class

jltha committed:
Add null checker to remove warning

jltha committed:
Change to static method call as method is static

jltha committed:
Remove unused line

jltha committed:
Add more @@author tags

jltha committed:
Add @@author tags

laiisaac committed:
Update developer guide

Add a section for study session commands in instructions for manual testing

jltha committed:
Update PPP

jltha committed:
Update PPP

jltha committed:
Update PPP

BradenTeo committed:
Add sequence diagram for parsing user input for edit command

BradenTeo committed:
Replace previously added screenshots from IDE to screenshots from
command prompt

BradenTeo committed:
Update PPP

BradenTeo committed:
Fix activation bar mistakes in sequence diagram

BradenTeo committed:
Update TimerClassDiagram.png to be of the standardised colour

jltha committed:
Fix typo in documentation

laiisaac committed:
Update PPP

laiisaac committed:
Merge pull request #159 from BradenTeo/branch-Minor-fixes

Braden Documentation and Minor fixes

jltha committed:
Merge branch 'master' into branch-v2.1-docs

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/seedu/sherpass/constant/Message.java

jltha committed:
Update testEdit.json

as identifer gets updated after testing using the save file

BradenTeo committed:
Merge pull request #161 from jltha/branch-v2.1-docs

Jun Lim v2.1 Documentation

thedarie committed:
Merge pull request #162 from laiisaac/isaac-BugFixes

Isaac v2.1 documentation

BradenTeo committed:
Fix more issues for Edit Parse User Input sequence diagram

BradenTeo committed:
Merge branch 'master' into branch-v2.1-Documentation-fixes

BradenTeo committed:
Update sequence diagram picture

jltha committed:
Set theme jekyll-theme-cayman

jltha committed:
Set theme jekyll-theme-cayman

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

laiisaac committed:
Merge branch 'master' into isaac-BugFixes

thedarie committed:
Merge branch 'master' into daryl-DG

laiisaac committed:
Update developer guide

Update screenshots for UML diagrams in timer
Update table of contents

BradenTeo committed:
Rename one sequence diagram, and remove one unused diagram

BradenTeo committed:
Add colour to <<class>> header in sequence diagrams
Fix activation bar issues in some sequence diagrams

BradenTeo committed:
Refactor PPP name to be in lowercase

thedarie committed:
Add color to class tag for sequence diagrams

thedarie committed:
Update PPP

thedarie committed:
Change incrementDate to be private

jltha committed:
Update SD Diagrams for showing schedule in DG

laiisaac committed:
Update developer guide

laiisaac committed:
Merge pull request #164 from thedarie/daryl-DG

Daryl v2.1 documentations

BradenTeo committed:
Merge branch 'BradenTeo-branch-v2.1-Documentation-fixes'

BradenTeo committed:
Merge branch 'master' into branch-v2.1-Documentation-fixes

BradenTeo committed:
Minor edit

jltha committed:
Update PPP

BradenTeo committed:
Merge branch 'master' of
https://github.com/AY2122S2-CS2113T-T09-1/tp

BradenTeo committed:
Merge branch 'master' into branch-v2.1-Documentation-fixes

laiisaac committed:
Merge pull request #166 from laiisaac/isaac-BugFixes

Isaac v2.1 documentation

jltha committed:
Merge branch 'master' into branch-v2.1-docs

BradenTeo committed:
Merge branch 'master' of
https://github.com/AY2122S2-CS2113T-T09-1/tp

BradenTeo committed:
Merge branch 'master' into branch-v2.1-Documentation-fixes

BradenTeo committed:
Merge pull request #163 from BradenTeo/branch-v2.1-Documentation-fixes

Braden v2.1 Documentation fixes

thedarie committed:
Fix PPP format

jltha committed:
Merge branch 'master' into branch-v2.1-docs

# Conflicts:
#	docs/images/ShowScheduleForADateSDPartOne.png
#	docs/puml/ShowScheduleForADateSDOne.puml

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

jltha committed:
Fix typo in teammate's github profile link

laiisaac committed:
Merge branch 'master' into isaac-BugFixes

jltha committed:
Merge pull request #165 from jltha/branch-v2.1-docs

Jun Lim v2.1 Doc fixes

thedarie committed:
Add target audience and purpose for DG

thedarie committed:
Change header to a more appropriate name

laiisaac committed:
Update screenshots in user guide for add and delete command

laiisaac committed:
Update user guide to fix formatting issue

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

laiisaac committed:
Merge branch 'master' into isaac-BugFixes

jltha committed:
Update formatting of PPP

thedarie committed:
Edit target audience section to be similar with UG

thedarie committed:
Merge pull request #168 from thedarie/daryl-DG

Daryl PPP formatting

yuzhongng committed:
Update yuzhong PPP

removes the bolding of header word

laiisaac committed:
Update screenshot in developer guide

yuzhongng committed:
Update User Guide

Replace all the timetable screenshot to be from cmd terminal instead of ide terminal

yuzhongng committed:
Replace screenshot to be from cmd terminal

yuzhongng committed:
Replace screenshot to be from cmd terminal

yuzhongng committed:
Replace screenshot to be from cmd terminal

yuzhongng committed:
Replace screenshot to be from cmd terminal

yuzhongng committed:
Replace screenshot to be from cmd terminal

yuzhongng committed:
Merge branch 'master' into yuzhong-Documentation

laiisaac committed:
Update more screenshots in user guide

jltha committed:
Merge pull request #170 from jltha/branch-v2.1-docs

Jun Lim PPP update

yuzhongng committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp into yuzhong-Documentation

jltha committed:
Fix formatting issue in PPP

laiisaac committed:
Minor updates to PPP

laiisaac committed:
Merge pull request #172 from yuzhongng/yuzhong-Documentation

Yuzhong documentation

laiisaac committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

jltha committed:
Add @@author tags

laiisaac committed:
Merge branch 'master' into isaac-BugFixes

laiisaac committed:
Merge pull request #169 from laiisaac/isaac-BugFixes

Isaac v2.1 documentation

jltha committed:
Add page break to DG

jltha committed:
Merge branch 'master' into branch-v2.1-docs

# Conflicts:
#	docs/DeveloperGuide.md

yuzhongng committed:
Update User Guide to fit the formatting of PDF file

yuzhongng committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp into yuzhong-Documentation

BradenTeo committed:
Update PPP

jltha committed:
Add more page breaks

yuzhongng committed:
update UG format

yuzhongng committed:
Merge pull request #173 from jltha/branch-v2.1-docs

Jun Lim v2.1 doc fixes

yuzhongng committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-1/tp

yuzhongng committed:
Merge branch 'master' into yuzhong-Documentation

BradenTeo committed:
Merge pull request #175 from BradenTeo/branch-Update-PPP

Braden Update PPP

laiisaac committed:
Add blank lines between images to improve formatting

laiisaac committed:
Add page break into PPP

BradenTeo committed:
Merge pull request #176 from laiisaac/isaac-BugFixes

Isaac v2.1 documentation

BradenTeo committed:
Add page break to PPP

BradenTeo committed:
Remove unnecessary element

laiisaac committed:
Merge pull request #174 from yuzhongng/yuzhong-Documentation

Yuzhong documentation

thedarie committed:
Fix typo in hyperlink

jltha committed:
Fix typo

thedarie committed:
Merge pull request #178 from thedarie/daryl-DG

Daryl Fix typo

yuzhongng committed:
Add author tag

yuzhongng committed:
Update PPP

BradenTeo committed:
Minor issue in DG

jltha committed:
Merge pull request #177 from BradenTeo/branch-Update-PPP

Braden Update PPP

jltha committed:
Merge pull request #179 from yuzhongng/yuzhong-Documentation

Yuzhong documentation

thedarie committed:
Update URL in PPP

thedarie committed:
Fix typo in Braden's PPP

thedarie committed:
Merge pull request #180 from thedarie/daryl-DG

Daryl Update PPP

thedarie committed:
Fix URL in PPP

thedarie committed:
Merge pull request #181 from thedarie/daryl-DG

Daryl Fix URL in PPP again

jltha committed:
Update UG

1) Fix typo
2) Update release link
3) Add definition to terminal

jltha committed:
Merge branch 'master' into branch-v2.1-docs

BradenTeo committed:
Update Quick Start

jltha committed:
Remove sentence for greater clarity in UG

jltha committed:
Merge pull request #182 from jltha/branch-v2.1-docs

Jun Lim UG updates

BradenTeo committed:
Merge branch 'master' into branch-UG-fix-for-CS2101

BradenTeo committed:
Merge pull request #183 from BradenTeo/branch-UG-fix-for-CS2101

Braden Update Quick Start

BradenTeo committed:
Add page breaks to UG

jltha committed:
Merge pull request #184 from BradenTeo/branch-UG-fix-for-CS2101

Braden Add page breaks to UG

BradenTeo committed:
Fix page break for UG

BradenTeo committed:
Merge pull request #185 from BradenTeo/branch-UG-fix-for-CS2101

Braden Fix page break for UG
